src/game/game_init.c

```diff
@@ -646,25 +644,13 @@ void read_controller_inputs(s32 threadID) {
             controller->stickMag = 0;
         }
     }
-
-    // For some reason, player 1's inputs are copied to player 3's port.
-    // This potentially may have been a way the developers "recorded"
-    // the inputs for demos, despite record_demo existing.
-    gPlayer3Controller->rawStickX = gPlayer1Controller->rawStickX;
-    gPlayer3Controller->rawStickY = gPlayer1Controller->rawStickY;
-    gPlayer3Controller->stickX = gPlayer1Controller->stickX;
-    gPlayer3Controller->stickY = gPlayer1Controller->stickY;
-    gPlayer3Controller->stickMag = gPlayer1Controller->stickMag;
-    gPlayer3Controller->buttonPressed = gPlayer1Controller->buttonPressed;
-    gPlayer3Controller->buttonReleased = gPlayer1Controller->buttonReleased;
-    gPlayer3Controller->buttonDown = gPlayer1Controller->buttonDown;
 }
 
 /**
  * Initialize the controller structs to point at the OSCont information.
  */
 void init_controllers(void) {
-    s16 port, cont;
+    s16 port, cont, lastUsedPort;
```

Comment:
another thing, what happens if a controller isn't plugged in at all? this should default to 0 to be safe so that this gets initialized properly

Reply:
It's initialized in the loop.